tests/test_router_registration.py

```diff
@@ -125,9 +126,20 @@ def test_register_router_on_itself() -> None:
 
 
 def test_route_handler_method_view() -> None:
-    @get(path="/test")
+    @get(path="/root")
     def handler() -> None:
         ...
 
-    router = Router(path="/", route_handlers=[MyController, handler])
-    assert router.route_handler_method_view[handler] == "/test"
+    first_router = Router(path="/first", route_handlers=[MyController, handler])
+    second_router = Router(path="/second", route_handlers=[MyController, handler])
+
+    # Test routers.
+    assert first_router.route_handler_method_view[handler.fn.__qualname__] == ["/first/root"]  # type: ignore
+    assert second_router.route_handler_method_view[MyController.get_method.fn.__qualname__] == ["/second/test"]  # type: ignore
```

Comment:
So using `__qualname__` is 99% sufficent but there are cases where it will not work as intended. Consider the following case:

```python
from starlite import put, post

def _handler(data: dict) -> dict:
    ...

put_handler = put("/")(handler)
post_handler = post("/")(handler)
```

Obviously this is a contrived example. The point though is that a truely unique identifier should consider a combination of the handler and fn qualnames. Hence my original example. 

Im leaving it to you to decide on the final implementation.

Reply:
2 route handler instances are decorating the same `fn` so their routes are adding up in the same route list, see the updates test case. I think this is fine. The point of this is to make it easy for the decorator to resolve all routes that goes through that `fn` so that the developer can add custom functionality based on these routes.

Reply:
ok, as i said - i leave it to you to decide on the final implementation. please merge when ready.